### ISSUE TITLE
🐛 BUG: `wrangler init --yes` leads to interactive prompt

### ISSUE THREAD

**Issue by samford** [cloudflare/workers-sdk#3286]
### Which Cloudflare product(s) does this pertain to?

Wrangler

### What version of `Wrangler` are you using?

3.0.0

### What operating system are you using?

Mac

### Describe the Bug

Hi, I'm a maintainer for the Homebrew package manager (also a `wrangler` user) and I was working on updating our formula to version 3.0.0 but I'm encountering a bug with the `--yes` option of `wrangler init`.

[Our test runs `wrangler init --yes`](https://github.com/Homebrew/homebrew-core/blob/46c97b044557305ef4295b3571db2ad3fc0725e0/Formula/cloudflare-wrangler2.rb#L32-L38), with the expectation that this will not produce any interactive prompts. This worked fine in previous versions but I'm now seeing the following with 3.0.0:

```
$ wrangler init --yes
 ⛅️ wrangler 3.0.0
------------------
Using npm as package manager.
▲ [WARNING] The `init` command is no longer supported. Please use `npm create cloudflare@2` instead.

  The `init` command will be removed in a future version.


Running `npm create cloudflare@2`...
Need to install the following packages:
  create-cloudflare@2.0.7
Ok to proceed? (y)
```

The warning states that `init` is no longer supported but `wrangler --help` still lists `wrangler init` as a command. I figured I would flag this `--yes` issue if it's something that can/should be fixed.

Otherwise, is there something more appropriate that we could/should be doing to test `wrangler` as a package manager? This would preferably be something that involves some basic functionality without depending on remote requests, authentication, etc. `wrangler init` was fine for our purposes but we may have to fall back to `wrangler --version` (which we try to avoid) if `init` will be removed in the future and there isn't a suitable alternative command to test.

**Comment by penalosa** [comment]
Thanks for raising this! The `--yes` flag not being respected is a regression in v3 that we'll be fixing in a future version. However, as the warning notes, `wrangler init` will be deprecated in a future version (in favour of `npm create cloudflare`), and so it's probably not a good long term way to test a Wrangler installation. I think `wrangler --version` is probably the best option for now—a lot of Wrangler's functionality relies on remote calls to the Cloudflare API, or expects user interactive.

As a side note, I notice that the brew package is called `cloudflare-wrangler2`—is there any way that could be changed to `wrangler`? It could be fairly confusing to users to have Wrangler v3 installed by a package called `wrangler2`

**Comment by samford** [comment]
Thanks for confirming. Due to the `--yes` behavior in 3.0.0, I think we'll have to fall back to `wrangler --version` in this version and we'll probably end up having to stick with that. After looking at the existing `wrangler` commands, it does appear that everything else (excluding `init`) involves network requests or some interactivity.

-----

Regarding the `cloudflare-wrangler2` formula name, I agree that it's not ideal but there are technical reasons for it. Homebrew has had a [`wrangler` formula](https://github.com/Homebrew/homebrew-core/blob/57f4d47f9dbc493ef45d53479e2726afb1d70b80/Formula/wrangler.rb) since at least 2009, so we had to use [`cloudflare-wrangler`](https://github.com/Homebrew/homebrew-core/blob/57f4d47f9dbc493ef45d53479e2726afb1d70b80/Formula/cloudflare-wrangler.rb) for the initial Rust 1.x versions. When the JavaScript 2.0.0 version was released, we had to create a separate formula for it, as both the Rust and JS versions were being maintained and there were users of both. The JS version was in the cloudflare/wrangler2 repository (though the command was still `wrangler`), so we followed the upstream naming convention and named the formula `cloudflare-wrangler2`.

At this point, we have a deprecated `cloudflare-wrangler` formula and `cloudflare-wrangler2` formula, so we can't rename the latter to the former and are probably unlikely to do so until after 2023-08-01 (when v1 becomes officially unsupported by Cloudflare). In the interim time, I've suggested renaming the formula to either `cloudflare-wrangler3` or something unversioned like `cloudflare-wrangler-js` until the Rust formula is eventually deleted and we can even consider the possibility of using `cloudflare-wrangler` for this formula.

It's not an ideal situation but hopefully that explains why this formula is named the way that it is and why we can't easily rename it to `cloudflare-wrangler` (let alone `wrangler`).

**Comment by penalosa** [comment]
@samford Thanks for clarifying the naming, and that makes sense. I don't know what Homebrew's policy is on re-naming formulas, but would something like renaming `cloudflare-wrangler` to `cloudflare-wrangler-legacy` and `cloudflare-wrangler2` to `cloudflare-wrangler` work?

**Comment by WalshyDev** [comment]
Personal addition, I use `wrangler init -y` all the time when creating new projects.

I haven't moved to v3 yet so this luckily hasn't impacted me so far but, it would be good to know what I'm supposed to do. The warning says to do `npm create cloudflare@2` but doing `npm create cloudflare@2 -y` just brings an interactive prompt, going through doesn't result in the same thing either (I can't have a vitest test created)

In v2, `init` was made to be really powerful, I'd personally love for that to stay or c3 to do the exact same (right now at the very least, it's missing non-interactive and testing).

Also small nit, wrangler.toml from `create cloudflare` is interesting... I'm gonna be deleting 80% of it every time. I'd personally just rather a comment saying like "see the docs for information on how to configure your wrangler.toml: <docs link>"
I can see the current one being quite overwhelming.

**Comment by RamIdeas** [comment]
The `init` command with the `-y` flag should be non-interactive since this fix https://github.com/cloudflare/workers-sdk/pull/3359 – can you confirm this fix works for your use-case?

**Comment by samford** [comment]
> The `init` command with the `-y` flag should be non-interactive since this fix #3359 – can you confirm this fix works for your use-case?

Unfortunately this still leads to the same interactive prompt on 3.1.1 for me:

```
$ wrangler init -y
 ⛅️ wrangler 3.1.1
------------------
Using npm as package manager.
▲ [WARNING] The `init` command is no longer supported. Please use `npm create cloudflare@2 -- --wrangler-defaults` instead.

  The `init` command will be removed in a future version.


Running `npm create cloudflare@2 -- --wrangler-defaults`...
Need to install the following packages:
  create-cloudflare@2.0.10
Ok to proceed? (y)
```
